### PR TITLE
CONTRIBUTING.md: Update commit guidelines

### PR DESCRIPTION
**Pull request by ejona86**
It came up in #11073, and I saw it could use a little updating. Notably, I'm linking to a guide to what Git commit messages should look like. I also tried to make the language less heavy-handed and demanding.

-----

I do think there's more that can be done, but I think this is a step in the right direction. It's obvious the IntelliJ style mention is out-of-date, for example.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,43 +30,36 @@ style configurations are commonly useful. For IntelliJ 14, copy the style to
...
+  or API change, consider starting with a [gRFC
+  proposal](https://github.com/grpc/proposal).
+
+- Follow [typical Git commit message](https://cbea.ms/git-commit/#seven-rules)
+  structure. Have a good **commit description** as a record of **what** and
+  **why** the change is being made. Link to a GitHub issue if it exists. The
+  commit description makes a good PR description and is auto-copied by GitHub if
+  you have a single commit when creating the PR.
+
+  If your change is mostly for a single module (other module changes are
```
Comment (larry-safran) [inline CONTRIBUTING.md:49]: adding `i.e.` or something similar to `other module changes` would make it clear that it is a clarification and read more easily.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,43 +30,36 @@ style configurations are commonly useful. For IntelliJ 14, copy the style to
...
+  trivial), prefix your commit summary with the module name changed. Instead of
+  "Add HTTP/2 faster-than-light support to gRPC Netty" it is more terse as
+  "netty: Add faster-than-light support".
+
+- Don't fix code style and formatting unless you are already changing that line
+  to address an issue. If you do want to fix formatting or style, do that in a
+  separate PR.
+
+- Unless your PR is trivial, you should expect there will be reviewer comments
+  that you'll need to address before merging. Make review changes as additional
```
Comment (larry-safran) [inline CONTRIBUTING.md:59]: "review changes" is nice and short, but feels like someone could get confused what it means (though the more I think about it, the less likely it seems).  "changes in response to reviewer comments" might be too verbose.

**Review by larry-safran — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (ejona86) [reply CONTRIBUTING.md:49]: I had noticed it wasn't great as well, but then said "meh" to myself. I do think `i.e.` would help a lot, although `e.g.,` is more accurate. So I'll use that.

File: CONTRIBUTING.md (hunk 2 above)
Comment (ejona86) [reply CONTRIBUTING.md:59]: Hmm... yeah, I see what you mean. "Address comments with additional commits..."? "Use additional commits for changes so the..."? "for needed changes"? "for necessary changes"? Does "changes" need an adjective?

**Pull request merged**